stuffer/s2n_stuffer.c

```diff
@@ -216,6 +216,7 @@ int s2n_stuffer_erase_and_read(struct s2n_stuffer *stuffer, struct s2n_blob *out
     GUARD(s2n_stuffer_skip_read(stuffer, out->size));
 
     void *ptr = stuffer->blob.data + stuffer->read_cursor - out->size;
+    /* cppcheck-suppress sizeofDereferencedVoidPointer */
```

Comment:
Does cppcheck operate on a post-macro input? If so, we should think about adding this suppression to the macro itself instead.

Reply:
Looking at where this warning would be triggered, it should only be happening when CBMC is enabled:

https://github.com/awslabs/s2n/blob/4ba1831705ad9339fb6f40d9a0bf5944f916e6b8/utils/s2n_safety.h#L357
https://github.com/awslabs/s2n/blob/4ba1831705ad9339fb6f40d9a0bf5944f916e6b8/utils/s2n_safety.h#L114
https://github.com/awslabs/s2n/blob/4ba1831705ad9339fb6f40d9a0bf5944f916e6b8/error/s2n_errno.h#L286

The non-CBMC definition just checks to see if it's `NULL`:

https://github.com/awslabs/s2n/blob/4ba1831705ad9339fb6f40d9a0bf5944f916e6b8/error/s2n_errno.h#L293

Reply:
I couldn't get it to work in the macro, it might be removing the comment when it expands the macro

Reply:
https://trac.cppcheck.net/ticket/3537